### PR TITLE
DevDB - Geocode applications during build

### LINKED ISSUE #536: DevDB - geocode job applications during build
The DevDB datasync action currently does a few things

- call data library step for three socrata datasets
  - one hpd dataset
  - dob permits issued
  - dob bis job applications (legacy system that seemingly still gets some use - dataset is regularly updated)
- geocode
  - for two hpd datasets, import datasets from edm-recipes, geocode and the archive them with data library. The second one is one that we have only manually loaded once
  - for applications data, another recipes dataset is imported - dob now applications. This is the newer system for filing applications with DOB. It is available on socrata, but we get a custom version via ftp from dob (with more columns than what's available on socrata). This is appended to the bis applications, geocoded, and exported as one dataset (dob_geocode_results)

Logic in DevDB build then imports ALL of these datasets - 2 hpd, 2 job applications - both ungeocoded and geocoded. Geocoded results are then joined based on runtime-generated ids of the input datasets line up with the geocode results. Which they should for  This means that if datasync runs, then dob_now data is loaded manually, and then devdb is built, thousands of records will join erroneously.

### PR DESCRIPTION
**Pull request by fvankrieken**
Resolves #536. Related to #523 as well

A few main changes
- removal of geocoding applications during datasync action. Datasync action still geocodes hpd data which is archived as first step of job
- call to geocode_dob.py during build. Instead of being aimed at a more raw concatenation of bis and now data, it's aimed at _init_devdb, which effectively has the same cleaning done on it upstream that was being done in the sql query in this file before
- little bit of sql cleanup

I tested this by making an intermediate table - persisted the DRAFT table defined in cte in _geo.sql - from both the archived geocode results and by using the new geocoded _init table, and ensuring they were completely identical (`select * from a EXCEPT select * from b`, and vice-versa)

### CODE REVIEW

File: dcpy/utils/geocode.py
Comment (fvankrieken) [inline dcpy/utils/geocode.py:1]: This isn't going to actually be part of this PR, but late yesterday just wanted to start working on moving geosupport to more usable, typed classes. Would love that to be part of python-geosupport repo at some point

**Comment by fvankrieken** [comment]
[Passing build](https://github.com/NYCPlanning/data-engineering/actions/runs/7575918809)

File: .github/workflows/developments_datasync.yml [hunk 1]
```diff
@@ -5,17 +5,14 @@ on:
     - cron: "0 0 * * MON"
   workflow_dispatch:
     inputs:
-      version:
-        description: "Would you like to specify a version for dob_jobapplications?"
```
Comment (alexrichey) [inline .github/workflows/developments_datasync.yml:9]: Such polite phrasing. A little sad to see it go

File: products/developments/python/geocode_dob.py [hunk 2]
```diff
@@ -35,15 +29,15 @@ def geocode(input):
...
-                borough=borough,
+                street_name=input["address_street"],
+                house_number=input["address_numbr"],
+                borough=input["boro"],
                 mode="tpad",
             )
             geo = parse_output(geo)
             geo.update(
                 dict(
-                    uid=uid,
```
Comment (alexrichey) [inline products/developments/python/geocode_dob.py:46]: Does changing these keys from uid to id change anything downstream?

Comment (fvankrieken) [reply products/developments/python/geocode_dob.py:46]: Yup - just used within the build as a unique row identifier for _init_devdb and downstream tables. They don't make it to export though. You'll see the same replacement in the sql files

**Review by alexrichey — APPROVED** [review]
Awesome

**Pull request merged**